tests/test_main.py

```diff
@@ -114,3 +116,19 @@ def test_custom_replay_file(monkeypatch, mocker, user_config_file) -> None:
         '.',
         'custom-replay-file',
     )
+
+
+def test_patch_import_path_for_repo():
+    """Test the _patch_import_path_for_repo context manager."""
+    original_sys_path = sys.path[:]
+    repo_dir = '/fake/repo/path'
+
+    patch = _patch_import_path_for_repo(repo_dir)
+    patch.__enter__()
+    assert sys.path[-1] == repo_dir
+    patch.__exit__(None, None, None)
+    assert sys.path == original_sys_path
```

Comment:
`.__enter__()` and `.__exit__()` are part of Python's context manager protocol; using `_patch_import_path_for_repo()` as a context manager would make this simpler:

```suggestion
    with _patch_import_path_for_repo(repo_dir):
        assert sys.path[-1] == repo_dir
    assert sys.path == original_sys_path
```